File: src/packets/PacketS2CZoneMute.js
const PacketS2C = require("./PacketS2C");

/**
 * Server -> Client
 * ID = 0x65
 * Zone Mute
 * Sends a zone's mute state
 * Data:
 *      (Unsigned Char) Controller ID
 *      (Unisgned Char) Zone ID
 *      (Unsigned Char) Muting
 */
class PacketS2CZoneMute extends PacketS2C {
    constructor(ctrlrID, zoneID, muted) {
        super();

        this._buffer.writeUInt8(ctrlrID);
        this._buffer.writeUInt8(zoneID);
        this._buffer.writeUInt8(muted ? 0x01 : 0x00);
    }

    getID() {
        return 0x65;
    }
}

module.exports = PacketS2CZoneMute;

File: src/RNetPi.Core/Packets/PacketS2CZoneMute.cs
namespace RNetPi.Core.Packets;

/// <summary>
/// Server -> Client
/// ID = 0x65
/// Zone Mute
/// Sends a zone's current mute state
/// Data:
///     (Unsigned Char) Controller ID
///     (Unsigned Char) Zone ID
///     (Unsigned Char) Muted
/// </summary>
public class PacketS2CZoneMute : PacketS2C
{
    public const byte ID = 0x65;

    public PacketS2CZoneMute(byte controllerID, byte zoneID, bool muted)
    {
        Writer.Write(controllerID);
        Writer.Write(zoneID);
        Writer.Write((byte)(muted ? 1 : 0));
    }

    public override byte GetID() => ID;
}